### PR TITLE
Check to make sure migration name is valid

### PR DESCRIPTION
**Pull request by AngelOnFira**
I created a migration named `initial-tables`, however, I had to change it since it wasn't valid as a Rust module name.

image

I'm sure there are some other characters that should be filtered out, but I didn't look too deep. Also, I created a custom error for this, but I'm not super sure if that's the cleanest place to put it. I can make any changes necessary :)

### CODE REVIEW

**Comment by tyt2y3** [comment]
Hi @AngelOnFira welcome back!

File: sea-orm-cli/src/commands.rs [hunk 1]
```diff
@@ -253,6 +253,14 @@ pub fn run_migrate_command(
             return Ok(());
         }
         Some(MigrateSubcommands::Generate { migration_name }) => {
+            // Make sure the migration name doesn't contain any characters that
+            // are invalid module names in Rust.
+            if migration_name.contains('-') {
+                return Err(Box::new(MigrationCommandError::InvalidName(
+                    "Invalid character used in migration name".to_string(),
+                )));
+            }
```
Comment (billy1624) [inline sea-orm-cli/src/commands.rs:262]: Hey @AngelOnFira, sorry for the delay. I think the error message can be more specific? E.g. hyphen `-` cannot be used in migration name

**Review by billy1624 — COMMENTED** [review]
One more nitpick, could we have a test case for this new behaviour? I want this behaviour to be checked and persisted in years to come :)
Thank you so much for the great contributions!

File: sea-orm-cli/src/commands.rs (hunk 1 above)
Comment (billy1624) [reply sea-orm-cli/src/commands.rs:262]: Such as...

```suggestion
            // Make sure the migration name doesn't contain any characters that
            // are invalid module names in Rust.
            if migration_name.contains('-') {
                return Err(Box::new(MigrationCommandError::InvalidName(
                    "Hyphen `-` cannot be used in migration name".to_string(),
                )));
            }
```

**Review by billy1624 — COMMENTED** [review]
Hey @AngelOnFira, I'll continue the work on https://github.com/SeaQL/sea-orm/pull/1155 :)

**Pull request closed without merging**